src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        List<T> result = new ArrayList<>();
+
+        if (type == Bulldozer.class) {
+            List<Bulldozer> bulldozers = new BulldozerProducer().get();
+            result.addAll((List<T>) bulldozers);
+        }
+        if (type == Excavator.class) {
+            List<Excavator> excavators = new ExcavatorProducer().get();
+            result.addAll((List<T>) excavators);
+        }
+        if (type == Truck.class) {
+            List<Truck> trucks = new TruckProducer().get();
+            result.addAll((List<T>) trucks);
```

Comment:
you can avoid calling the getAll method by assigning the result of the get method to a variable `result`